src/QMCWaveFunctions/tests/test_MO_spinor.cpp

```diff
@@ -645,8 +673,104 @@ void test_lcao_spinor_excited()
   }
 }
 
+void test_lcao_spinor_ion_derivs()
+{
+  app_log() << "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n";
+  app_log() << "!!!! LCAO SpinorSet from HDF (ion derivatives) !!!!\n";
+  app_log() << "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n";
+
+  using ValueType = SPOSet::ValueType;
+  using RealType  = SPOSet::RealType;
+  Communicate* c  = OHMMS::Controller;
+
+  ParticleSetPool ptcl = ParticleSetPool(c);
+  auto ions_uptr       = std::make_unique<ParticleSet>(ptcl.getSimulationCell());
+  auto elec_uptr       = std::make_unique<ParticleSet>(ptcl.getSimulationCell());
+  ParticleSet& ions_(*ions_uptr);
+  ParticleSet& elec_(*elec_uptr);
+
+  ions_.setName("ion");
+  ptcl.addParticleSet(std::move(ions_uptr));
+  ions_.create({2});
+
+  ions_.R[0][0]        = 0.10000000;
+  ions_.R[0][1]        = 0.20000000;
+  ions_.R[0][2]        = 0.30000000;
+  ions_.R[1][0]        = -0.30000000;
+  ions_.R[1][1]        = -0.20000000;
+  ions_.R[1][2]        = -0.10000000;
+  SpeciesSet& ispecies = ions_.getSpeciesSet();
+  int hIdx             = ispecies.addSpecies("H");
+  ions_.update();
+
+  elec_.setName("elec");
+  ptcl.addParticleSet(std::move(elec_uptr));
+  elec_.create({1});
+  elec_.R[0][0]  = 0.01;
+  elec_.R[0][1]  = -0.02;
+  elec_.R[0][2]  = 0.03;
+  elec_.spins[0] = 0.6;
+  elec_.setSpinor(true);
+
+  SpeciesSet& tspecies       = elec_.getSpeciesSet();
+  int upIdx                  = tspecies.addSpecies("u");
+  int chargeIdx              = tspecies.addAttribute("charge");
+  tspecies(chargeIdx, upIdx) = -1;
+
+
+  elec_.addTable(ions_);
+  elec_.update();
+
+  const char* particles = "<tmp> \
```

Comment:
Please modernize all the XML blocks with `R"XML(...)XML"`
See https://github.com/QMCPACK/qmcpack/blob/97b98a8ef466ade8ce0e81348243529488633ed6/src/QMCWaveFunctions/tests/test_J1_bspline.cpp#L97

Reply:
done